packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -835,6 +839,45 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         this.setState({ searchString: event.target.value ?? '' });
     }
 
+    private handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (
+            event.key === 'Enter' &&
+            this.state.searchString.trim() &&
+            event.target instanceof HTMLInputElement &&
+            event.target.classList.contains('MuiInputBase-input')
+        ) {
+            this.addFilter(this.state.searchString.trim());
+            this.setState({ searchString: '' });
+        }
+    };
+
+    private addFilter = (filter: string) => {
+        this.setState(prevState => ({ filters: [...prevState.filters, filter] }), this.updateSearchFilters);
+    };
+
+    private removeFilter = (filter: string) => {
+        this.setState(
+            prevState => ({ filters: prevState.filters.filter(f => f !== filter) }),
+            this.updateSearchFilters
+        );
+    };
+
+    private updateSearchFilters = () => {
+        const filterExpressionsMap: { [key: number]: string[] } = {};
+        if (this.state.searchString) {
+            filterExpressionsMap[1] = [this.state.searchString]; // For greying out
+        }
+        if (this.state.filters.length > 0) {
+            filterExpressionsMap[4] = this.state.filters; // For filtering
```

Comment:
Could we use constants for these key values?
const DIMMED = 1; // DIMMED filter expression property key
const EXCLUDED = 4; // EXCLUDED filter expression property key

Reply:
Done!